src/groupeddataframe/grouping.jl

```diff
@@ -181,6 +197,319 @@ end
 Base.getindex(gd::GroupedDataFrame, idxs::Colon) =
     GroupedDataFrame(gd.parent, gd.cols, gd.groups, gd.idx, gd.starts, gd.ends)
 
+
+"""
+    groupindices(gd::GroupedDataFrame)
+
+Return a vector of group indices for each row of `parent(gd)`.
+
+Rows appearing in group `gd[i]` are attributed index `i`. Rows not present in
+any group are attributed `missing` (this can happen if `skipmissing=true` was
+passed when creating `gd`, or if `gd` is a subset from a larger [`GroupedDataFrame`](@ref)).
+"""
+groupindices(gd::GroupedDataFrame) = replace(gd.groups, 0=>missing)
+
+"""
+    groupvars(gd::GroupedDataFrame)
+
+Return a vector of column names in `parent(gd)` used for grouping.
+"""
+groupvars(gd::GroupedDataFrame) = _names(gd)[gd.cols]
+
+# Get grouping variable index by its name
+function _groupvar_idx(gd::GroupedDataFrame, name::Symbol, strict=true)
+    i = findfirst(==(name), groupvars(gd))
+    i === nothing && strict && error("$name is not a grouping column")
+    return i
+end
+
+"""
+    groupvalues([T::Type], gd::GroupedDataFrame)
```

Comment:
Do we still need this function? My idea was that `keys` (or `NamedTuple.(keys(gd))`) would replace it.

Reply:
That works for getting all the keys at once. There are also methods for getting values for a specific index and column, though, which I think could be useful for end users and are also used internally quite a bit.

Reply:
Well in that case I'd rather have `keys` return an indexable object (probably `<: AbstractVector`) so that you can extract the index for group `i`. Then you can also extract the column you want since it contains named tuples.